client/ipa-getkeytab.c

```diff
@@ -196,6 +196,21 @@ static int ipa_ldap_init(LDAP **ld, const char *ldap_uri)
 	return rc;
 }
 
+static int ipa_starttls_init(LDAP *ld, const char *ldap_uri)
+{
+    int rc = LDAP_SUCCESS;
+    char *startswith_ldap = NULL;
+
+    startswith_ldap = strstr("ldap://", ldap_uri);
```

Comment:
ldap:// can be only at the start of the string, pleae use strncmp(ldap_uri, "ldap://", 7)